README.md

```diff
@@ -123,6 +123,26 @@ let x = 0.0
 end
 ```
 
+Note that the let block is used here to create a scope to allow assignment of 
+`x` within `@byrow!`.
+
+`byrow!` also supports special syntax for allocating new columns to make
+`byrow!` more useful for data transformations. The syntax `@newcol
+x::Array{Int}` allocates a new column `:x` with an `Array` container with eltype
+`Int`. Note that the returned AbstractDataFrame includes these new columns, but
```

Comment:
AbstractDataFrame needs backquotes (or just say "data frame").

I find it a bit problematic that the new columns are not added to the original data frame, given that the macro is called `@byrow!`. Is there a technical reason for this?